cli/src/pcluster/cli/commands/configure/easyconfig.py

```diff
@@ -177,11 +183,20 @@ def configure(args):  # noqa: C901
         compute_resources = []
         for compute_resource_index in range(number_of_compute_resources):
             if scheduler != "awsbatch":
-                compute_instance_type = prompt(
-                    f"Compute instance type for compute resource {compute_resource_index+1} in {queue_name}",
-                    validator=lambda x: x in AWSApi.instance().ec2.list_instance_types(),
-                    default_value=default_instance_type,
-                )
+                while True:
+                    compute_instance_type = prompt(
+                        f"Compute instance type for compute resource {compute_resource_index+1} in {queue_name}",
+                        validator=lambda x: x in AWSApi.instance().ec2.list_instance_types(),
+                        default_value=default_instance_type,
+                    )
+                    if compute_instance_type not in [
+                        compute_resource["InstanceType"] for compute_resource in compute_resources
+                    ]:
+                        break
+                    print(
+                        f"Error: Instance type {compute_instance_type} cannot be specified for multiple compute "
```

Comment:
the bad part here is that by default the pcluster configure workflow is proposing the same instance type for all the compute resources. Let's have a list of predefined instance types (e.g.: t2.micro, t2.small, t2.medium, etc) and let's propose them in order if not already added in the previous compute resources.

Reply:
I have some concerns on have a list of predefined instance types.
Currently we are use free-tier-eligible instance types as default instance types(t2.micro in t2.micro free regions and t3.micro in t3.micro in regions there's no t2micro) If we are suggest other instance types, there will be extra costs on it
It may be hard to choose the the predefined instance types list. If users have many queues, should the instance types be in the order of  their on-demand price? I didn't see in the aws api have the option of sorting instance types by price. We may need some extra work on doing this
Do you have some idea on this?

Reply:
Let's keep the things as they are, if you tested it and it's a good UX we can proceed, thanks!